mindsdb/interfaces/storage/fs.py

```diff
@@ -48,10 +47,41 @@ class RESOURCE_GROUP:
 SERVICE_FILES_NAMES = (DIR_LOCK_FILE_NAME, DIR_LAST_MODIFIED_FILE_NAME)
 
 
+def compare_recursive(comparison: filecmp.dircmp) -> bool:
+    """Check output of dircmp and return True if the directories do not differ
+
+    Args:
+        comparison (filecmp.dircmp): dirs comparison
+
+    Returns:
+        bool: True if dirs do not differ
+    """
+    if comparison.left_only or comparison.right_only or comparison.diff_files:
+        return False
+    for sub_comparison in comparison.subdirs.values():
+        if compare_recursive(sub_comparison) is False:
+            return False
+    return True
+
+
+def compare_directories(dir1: str, dir2: str) -> bool:
+    """Compare two directories
+
+    Args:
+        dir1 (str): dir to compare
+        dir2 (str): dir to compare
+
+    Returns:
+        bool: True if dirs do not differ
+    """
+    dcmp = filecmp.dircmp(dir1, dir2)
```

Comment:
do you need to compare a content of files inside? 
As I see, it compares only time of the files by default
https://docs.python.org/3.11/library/filecmp.html#filecmp.cmpfiles

Reply:
`If shallow is true and the [os.stat()](https://docs.python.org/3.11/library/os.html#os.stat) signatures (file type, size, and modification time) of both files are identical, the files are taken to be equal.`
@ea-rus type, time and size, that should be enough